bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java

```diff
@@ -48,6 +56,7 @@
  * Handles communicating with a Daikin air conditioning unit.
  *
  * @author Tim Waterhouse - Initial Contribution
+ * @author Paul Smedley <paul@smedley.id.au> - Modifications to support Airbase Controllers
```

Comment:
You might consider creating a new class `AirbaseAcUnitHandler` extending the class `DaikinAcUnitHandler` and put all Airbase specific code into that class.  For specific code make or add in the `DaikinAcUnitHandler` methods protected and override them in the new class. For example with `pollStatus`.  You simply override this method and put the content of the method `pollAirbaseStatus` in the overridden method. That way you don't have to do checks on thing id and keep the code cleaner. In the Factory class you can simply instantiate with the specific handler.

Reply:
I see this one is flagged as 'might consider'. Can this one be a continuous improvement item for later? I have other things on the roadmap once the basic airbase changes are in git

Reply:
Yes I think that is ok.

Reply:
Thanks. I'll keep it in mind for future PRs